src/ioctls/vcpu.rs

```diff
@@ -1339,6 +1339,39 @@ impl VcpuFd {
         let kvm_run = self.kvm_run_ptr.as_mut_ref();
         kvm_run.immediate_exit = val;
     }
+
+    /// Returns the vCPU TSC frequency in KHz or an error if the host has unstable tsc.
+    ///
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn get_tsc_khz(&self) -> Result<u32> {
+        // Safe because we know that our file is a KVM fd and that the request is one of the ones
+        // defined by kernel.
+        let ret = unsafe { ioctl(self, KVM_GET_TSC_KHZ()) };
+        if ret >= 0 {
+            Ok(ret as u32)
+        } else {
+            Err(errno::Error::new(ret))
```

Comment:
Can you also add a negative test?

Reply:
Unfortunately, there is no way to reproduce the negative outcome of the function in a reliable way in the test. I think we would need mock structs for this purpose.

I updated the tests to check for machines where these capabilities are not present though.

Reply:
Can we test this with a faulty vcpu fd? Here is the test: https://github.com/rust-vmm/kvm-ioctls/blob/9d0fd1fbc22918fdd03c89ca888195a8ad297685/src/ioctls/vcpu.rs#L1910

Reply:
I tried using that example, PTAL.